pinot-core/src/main/java/org/apache/pinot/core/data/manager/InstanceDataManager.java

```diff
@@ -86,11 +86,17 @@ void addRealtimeSegment(String realtimeTableName, String segmentName)
       throws Exception;
 
   /**
-   * Removes a segment from a table.
+   * Removes a segment from a table but not dropping its data from server.
    */
   void removeSegment(String tableNameWithType, String segmentName)
       throws Exception;
 
+  /**
+   * Delete segment data from the server.
+   */
+  void dropSegment(String tableNameWithType, String segmentName)
```

Comment:
Consider renaming to `deleteSegmentData` to be more specific

Reply:
will rename removeSegment to offloadSegment() and call this one deleteSegment()